source/common/quic/client_connection_factory_impl.cc

```diff
@@ -7,12 +7,49 @@
 namespace Envoy {
 namespace Quic {
 
+// A wrapper class for quic::SessionCache.
+class QuicSessionCacheDelegate : public quic::SessionCache {
```

Comment:
Out of curiosity, why is this called "Delegate"? I might have expected it to be QuicSessionCacheImpl or some such (to suggest that it's a subclass of SessionCache) whereas Delegate suggests that it's a helper class for a SessionCache.

Reply:
This class is merely a wrapper over the real session cache object. It is needed because of object life time conflict. QuicCryptoClientConfig takes the ownership of a session cache object, but QuicCryptoClientConfig is owned by conn pool while the real session cache object is owned by QuicPersistentInfo object which is owned by cluster. To make the real session cache out-live conn pool, I added this wrapper to delegate all the interface calls to the real session cache object, but the wrapper itself only lives within the life time of a conn pool.

Reply:
Ah! I see. Ok, so let's add a class comment which explains the purpose of this class (the lifetime concerns). Also which the class delegates to the underlying session, it itself is not a delegate. I think it is, as you said, a wrapper. So how about "QuicSessionCacheWrapper"?